### PR TITLE
(chore) ‘apply filters’ becomes ‘(refine) search’

### PR DESCRIPTION
**Pull request by gazaston**
Some testing has revealed the button text "Apply filters" can be confusing.

We've changed it to 'Search' for the initial state, and 'Refine search' when a search has already been performed.

**Before**
before

**After (no search applied)**
after no search

**After (with search applied)**
after with search

### CODE REVIEW

File: spec/features/job_seekers_can_view_vacancies_spec.rb [hunk 1]
```diff
@@ -125,6 +125,28 @@
     expect(page).not_to have_content(I18n.t('jobs.no_jobs'))
   end
 
+  scenario 'Should have button with text \'Search\' if default search criteria are applied' do
+    Vacancy.__elasticsearch__.client.indices.flush
+    visit jobs_path
+
+    # expect(page).to have_content(I18n.t('buttons.apply_filters'))
```
Comment (despo) [inline spec/features/job_seekers_can_view_vacancies_spec.rb]: can we remove this please?

File: spec/features/job_seekers_can_view_vacancies_spec.rb [hunk 2]
```diff
@@ -125,6 +125,28 @@
...
+    expect(find('.button').value).to eq(I18n.t('buttons.apply_filters'))
+  end
```
Comment (despo) [inline spec/features/job_seekers_can_view_vacancies_spec.rb]: I wonder if we do need this scenario at all since the other scenario caters for both states. Maybe rewording the title eg `The search button text changes from 'Search' to 'Refine search' when the filters are applied`.

What do you think?

Comment (gazaston) [reply spec/features/job_seekers_can_view_vacancies_spec.rb]: Agreed 👍

File: spec/features/job_seekers_can_view_vacancies_spec.rb [hunk 3]
```diff
@@ -125,6 +125,27 @@
...
+  scenario 'The search button text changes from \'Search\' to \'Refine search\' when the filters are applied' do
+    Vacancy.__elasticsearch__.client.indices.flush
+    visit jobs_path
+
+    expect(find('.button').value).to eq(I18n.t('buttons.apply_filters'))
+  end
+
+  scenario 'Should have button with text \'Refine search\' if search criteria are applied' do
+    visit jobs_path
+
```
Comment (despo) [inline spec/features/job_seekers_can_view_vacancies_spec.rb]: Can remove lines 133-137. Then the one scenario verifies the existence of both `Search` and `Refine Search` after applying the criteria.
